docker/mcp/utils.py

```
@@ -0,0 +1,58 @@
+
+def clean_text(text: str) -> str:
+    """Clean scraped text by normalizing whitespace and newlines."""
+    if not text:
+        return ""
+    # Replace multiple newlines with a single newline
+    text = '\n'.join(line.strip() for line in text.splitlines() if line.strip())
+    # Replace multiple spaces with a single space
+    text = ' '.join(text.split())
+    return text
+
+def try_fetch_with_backoff(url: str, headers: Dict[str, str], attempts: int = 3, backoff_factor: int = 2) -> BeautifulSoup:
```

Pylint (3.3.7)

[convention] Line too long (123/100)

(C0301)